### DISCUSSION TITLE
Episode: Zack - Module Federation

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#66] [Announcements]
## Intro

Andrew: Hello, welcome to the devtools.fm podcast. This is a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin. 

Justin: Hey everyone!

Andrew: Today our guest is Zack Jackson, a principal engineer at Lululemon, maintainer of webpack, and co-creator of module federation. Today we'll take a deep dive into what exactly module federation is and how you can use it in your apps. Zack would you like to tell our audience a little more about yourself and how you got into creating developer tooling?

## Questions

- [x] What is module federation? What problems were you having that lead you creating this solution?
- [x] You got module federation merged into webpack itself instead of a building a competing tool. How was that process of getting such a new idea into webpack? Why did you choose webpack?
- [x] What sets of problems can module federation solve? I see it used to create a "micro frontend" architecture a lot, but how can it be used in traditional apps? 
- [x] What is software streaming in regards to module federation? How is it different than just requisition files from s3?
- [x] In one of your tweets you said that most deployment architecture is outdated. How does module federation and software streaming take us into the future of software deployment?
- [x] Does tree shaking work across module federation boundaries? How do I only expose part of a module?
- [x] Do types work?
- [x] Is there an easy way to have a local fallback? 
- [x] I've seen you tweeting about your federated next.js ssr plugin. What have been some of the challenges building that? Why does it need it's own plugin?
- [x] You're selling this plugin for a monthly fee? Selling software in this way seems less common these days. How has this been as a business model? Do you think more devs should sell their work like this?
- [x] Module federation is currently getting the spotlight in front end, but how could it be used in the backend?
- [x] What do you think the future of the web looks like? 

## Outro

Andrew: Well that’s it for this week’s episode of devtools.fm. Be sure to follow us on YouTube and wherever you consume your podcasts. Thanks for listening!

**Comment by ScriptedAlchemy** [comment]
https://github.com/module-federation/module-federation-examples
https://app.privjs.com/buy/packageDetail?pkg=@module-federation/nextjs-mf
https://app.privjs.com/buy/packageDetail?pkg=@module-federation/nextjs-ssr

**Comment by hipstersmoothie** [comment]
https://www.patterns.dev
https://reactjs.org/blog/2022/03/29/react-v18.html

**Comment by just-be-dev** [comment]
https://github.com/facebook/lexical
https://blog.maximeheckel.com/posts/the-power-of-composition-with-css-variables/
https://shop.vectorfinesse.com/
